### PR TITLE
Add Publisher in Octomap Plugin

### PR DESCRIPTION
**Pull request by acfloria**
Add option to publish octomap in the ServiceCallback of the gazebo_octomap_plugin as discussed with @ffurrer .

### CODE REVIEW

**Comment by ffurrer** [comment]
ok to test

File: rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp [hunk 1]
```diff
@@ -63,14 +64,14 @@ bool OctomapFromGazeboWorld::ServiceCallback(
   res.map.header.frame_id = "world";
   res.map.header.stamp = ros::Time(now.sec, now.nsec);
 
-  if (octomap_msgs::binaryMapToMsgData(*octomap_, res.map.data)) {
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp]: +1

File: rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp [hunk 2]
```diff
@@ -38,6 +38,7 @@ void OctomapFromGazeboWorld::Load(physics::WorldPtr _parent,
   gzlog << "Advertising service: " << service_name << std::endl;
   srv_ = node_handle_.advertiseService(
       service_name, &OctomapFromGazeboWorld::ServiceCallback, this);
+  octomap_publisher_ = node_handle_.advertise<octomap_msgs::Octomap>("world/octomap", 1, true);
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp]: Can you define a string for the topic name and add an SDF tag and then use: 

```C++
getSdfParam<std::string>(_sdf, "octomapPubTopic", octomap_pub_topic_, octomap_pub_topic_);
```
(would be good to also have these for the service name).

File: rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp [hunk 3]
```diff
@@ -63,14 +64,14 @@ bool OctomapFromGazeboWorld::ServiceCallback(
...
-    res.map.binary = true;
-    res.map.resolution = octomap_->getResolution();
-  } else {
+  if (!octomap_msgs::binaryMapToMsg(*octomap_, res.map)) {
     ROS_ERROR("Error serializing OctoMap");
   }
-  std::cout << "Publishing Octomap." << std::endl;
+
+  if (req.publish_octomap) {
+    std::cout << "Publishing Octomap." << std::endl;
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp]: Switch to gzlog, here and above.

File: rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp (hunk 2 above)
Comment (acfloria) [inline rotors_gazebo_plugins/src/gazebo_octomap_plugin.cpp]: Changed for the topic and service name.

**Pull request merged**